### ISSUE TITLE
[Code Health] Use `undefined` instead of `null` where possible

### ISSUE THREAD

**Issue by gino-m** [google/ground-platform#459]
I couldn't find guidance on this in the Google TypeScript style (GTS), but TypeScript team advise avoiding `null` (https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined). I think applying this guidance consistently will simplify our code, since we won't need to consider both states (undefined and null), and we can use `?` optional type instead of (or in addition to) `| null`.

**Comment by gino-m** [comment]
I experimented a bit with replacing `null` with `undefined` and at first glance I don' think there's a big benefit to doing so in our codebase. APIs like Router (ParamMap) return null, requiring additional boilerplate to covert to undefined. Also, since we use RxJS throughout our code, we will need to add `| undefined` in generics anyway (`Observable<string | undefined>`), which doesn't really help much with readability. Closing this for now unless we come up with a better solution.
